### PR TITLE
chore: bump interfaces-go-ipfs-core

### LINKED ISSUE #167: Not compatible with `github.com/ipfs/kube@v0.18.1`
## Description

Projects that depend on ipfs/kubo@v0.18.1 cannot use ipfs/go-ipfs-http-client@v0.4.0. If we try, we get errors like:

```
go build .
# github.com/libp2p/go-libp2p-core/network
../../../go/pkg/mod/github.com/libp2p/go-libp2p-core@v0.20.1/network/rcmgr.go:166:27: network.NullResourceManager (type) is not an expression

go mod why github.com/libp2p/go-libp2p-core/network
# github.com/libp2p/go-libp2p-core/network
github.com/ipfs/go-ipfs-http-client
github.com/libp2p/go-libp2p-core/network
```

ipfs/go-ipfs-http-client depends on:

https://github.com/ipfs/go-ipfs-http-client/blob/171fcd55e3b743c38fb9d78a34a3a703ee0b5e89/go.mod#L12

Which depends on:

https://github.com/ipfs/go-ipfs-http-client/blob/171fcd55e3b743c38fb9d78a34a3a703ee0b5e89/go.mod#L15

But is archived and contains deprecated methods like:

https://github.com/libp2p/go-libp2p-core/blob/654214c1b3401c0363ef464ce9db7e1b04a7ef3f/network/rcmgr.go#L165-L166

## Potential next steps?

1. [Refactor ipfs/interface-go-ipfs-core](https://github.com/ipfs/interface-go-ipfs-core/blob/455c16d43cca0a2ea6949f71bf57c08d9560d5ac/go.mod#L11) to depend on github.com/libp2p/go-libp2p@v0.24.2, and use its core package instead of github.com/libp2p/go-libp2p-core@v0.8.6
2. Update ipfs/go-ipfs-http-client to use the new ipfs/interface-go-ipfs-core using github.com/libp2p/go-libp2p@v0.24.2

## Anything else

I wanted to surface an issue to the community to get it's corresponding guidance. Let me know what you think? :pray:

### PR DESCRIPTION
**Pull request by Jorropo**
Fixes #167. Closes #166. Closes #163.

### CODE REVIEW

**Comment by marten-seemann** [comment]
Related: #169. @Jorropo You'll need to make sure that go-libp2p-core doesn't appear in the dependencies (including transitive) at all.

**Review by marten-seemann — APPROVED** [review]

**Review by Jorropo — COMMENTED** [review]
Don't merge, still need tests

File: routing.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"context"
+	"encoding/json"
+	"errors"
+)
+
+type RoutingAPI HttpApi
+
+func (api *RoutingAPI) Get(ctx context.Context, k string) (out []byte, err error) {
+	resp, err := api.core().Request("dht/findpeer", k).Send(ctx)
```
Comment (hacdias) [inline routing.go:12]: There is `routing/get`: https://docs.ipfs.tech/reference/kubo/rpc/#api-v0-routing-get

Comment (hacdias) [reply routing.go:12]: See:
- https://github.com/ipfs/kubo/pull/9628
- https://github.com/ipfs/kubo/pull/9628
- https://github.com/ipfs/interface-go-ipfs-core/pull/104
- https://github.com/ipfs/kubo/pull/9629

**Pull request merged**